aopy/preproc/neuropixel.py

```diff
@@ -239,4 +240,41 @@ def sync_ts_data_timestamps(data, sync_timestamps):
         sync_data = data[not_crop_datapoints,:]
         sync_timestamps = sync_timestamps[not_crop_datapoints]
         
-    return np.squeeze(sync_data), sync_timestamps
\ No newline at end of file
+    return np.squeeze(sync_data), sync_timestamps
+
+def destripe_lfp_batch(lfp_data, save_path, sample_rate, bit_volts, batch_length = 180, dtype='int16'):
+    
+    '''
+    Destripe LFP data in each batch to save memory. The result is saved in save_path.
+    
+    Args:
+        lfp_data (nt, nch): LFP data. This should be a memory mapping array.
+        save_path (str): file path to save destriped lfp data
+        sample_rate (float): sampling rate in Hz
+        bit_volts (float): volt per bit
+        batch_length (int): batch size in second
+    

```

Comment:
I think it would be more useful/intuitive to set the batch length based on memory available. The input argument would be something like "memory_available" in GB, and then would calculate the batch length based on the number of channels and data type. This way the function would always run the fastest allowable based on the memory available. 

What do you think? If we do this, would we need a minimum batch length for the destriping function to work?